### PR TITLE
[#8] fact extraction

### PR DESCRIPTION
**Pull request by Samuel-Campbell**
STILL MISSING GIT LFS INTEGRATION FOR PICKLE FILES
AND NEED TO UPDATE __INIT__.py

### CODE REVIEW

File: .idea/JusticeAI.iml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (arekmano) [inline .idea/JusticeAI.iml]: remove the .idea files. add it to .gitignore

File: .idea/misc.xml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (arekmano) [inline .idea/misc.xml]: remove

File: .idea/modules.xml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (arekmano) [inline .idea/modules.xml]: remove

File: .idea/vcs.xml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (arekmano) [inline .idea/vcs.xml]: remove

File: src/nlp_service/feature_extraction/Documentation/clause_finder.json [hunk 5]
```diff
@@ -0,0 +1,128 @@
+{"data": [
```
Comment (arekmano) [inline src/nlp_service/feature_extraction/Documentation/clause_finder.json]: what am I looking at?

File: src/nlp_service/feature_extraction/Documentation/example.txt [hunk 6]
```diff
@@ -0,0 +1,33 @@
+{
```
Comment (arekmano) [inline src/nlp_service/feature_extraction/Documentation/example.txt]: again what?

File: src/nlp_service/feature_extraction/FactTree/LogicParser.py [hunk 7]
```diff
@@ -0,0 +1,172 @@
+import nltk
+from bllipparser import RerankingParser
+from bllipparser.ModelFetcher import download_and_install_model
+from feature_extraction.GlobalVariables.regex import Regex
+from feature_extraction.Models import compliment, predicate, clause
+
+
+class Tree:
+    model_dir = download_and_install_model('WSJ', '/tmp/models')
```
Comment (arekmano) [inline src/nlp_service/feature_extraction/FactTree/LogicParser.py:9]: this belongs in init.py

File: src/nlp_service/feature_extraction/FactTree/PropositionLogic.py [hunk 8]
```diff
@@ -0,0 +1,189 @@
+from feature_extraction.FactTree import LogicParser, WordStack
+from feature_extraction.Models import clause, predicate, compliment
+from feature_extraction.Models import proposition
+
+
+class Proposition():
```
Comment (arekmano) [inline src/nlp_service/feature_extraction/FactTree/PropositionLogic.py:6]: This file is GG when we switch to french

File: src/nlp_service/feature_extraction/WordVectors/word_vector.py [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+
+
+class WordVectors:
+    def __init__(self):
+        path = os.path.dirname(os.path.realpath(__file__))
+        self.pickle_path = os.path.join(path, 'word_vector.pickle')
+
+    #####################################
+    # LOAD MODEL FROM FILE
+    def load_glove_model_file(self, file):
```
Comment (arekmano) [inline src/nlp_service/feature_extraction/WordVectors/word_vector.py:13]: when is this used?

**Review by arekmano — CHANGES_REQUESTED** [review]

File: src/nlp_service/feature_extraction/FactTree/PropositionLogic.py (hunk 8 above)
Comment (Samuel-Campbell) [reply src/nlp_service/feature_extraction/FactTree/PropositionLogic.py:6]: nah there is a french corpus that can train the same model. Would need an adapter class maybe

File: src/nlp_service/feature_extraction/FactTree/LogicParser.py (hunk 7 above)
Comment (Samuel-Campbell) [reply src/nlp_service/feature_extraction/FactTree/LogicParser.py:9]: This downloads the model if it doesn't exist, else it loads it. It's 2 functions the name is deceiving

File: src/nlp_service/feature_extraction/WordVectors/word_vector.py (hunk 9 above)
Comment (Samuel-Campbell) [reply src/nlp_service/feature_extraction/WordVectors/word_vector.py:13]: for word2vec

File: src/nlp_service/.gitignore [hunk 10]
```diff
@@ -7,4 +7,7 @@ feature_extraction/WordVectors/*.pickle
 .coverage
 .idea/
 ./**/*.iml
-.idea/misc.xml 
+.idea/misc.xml
```
Comment (arekmano) [inline src/nlp_service/.gitignore]: should be .idea/*

**Review by arekmano — APPROVED** [review]

**Pull request merged**